python/cuml/dask/common/input_utils.py

```diff
@@ -212,14 +212,16 @@ def _extract_partitions(dask_obj, client=None):
 
     client = default_client() if client is None else client
 
+    dask_obj = client.persist(dask_obj)
     # dask.dataframe or dask.array
     if isinstance(dask_obj, dcDataFrame) or \
             isinstance(dask_obj, daskArray):
-        parts = futures_of(client.compute(dask_obj))
+        parts = client.compute(futures_of(dask_obj))
 
     # iterable of dask collections (need to colocate them)
     elif isinstance(dask_obj, Sequence):
-        parts = [futures_of(client.compute(a)) for a in dask_obj]
+        parts = [client.compute(futures_of(d))
+                 for d in dask_obj]
```

Comment:
What's the goal in this code? Do we want all the `Future`s that make up all Dask object or do we want a `Future` per Dask object? In either case I think we can simplify this a bit. Just trying to figure out which way we should go.

Reply:
It's the former. We want all `Future`s that make up the dask object, which in this case I believe are the partitions of that dask object

Reply:
Thanks. I think we can just do `parts = futures_of(dask_obj)` up top and drop the branches here. Though please double check in case I'm missing what you are after.

Reply:
Ah, I see. Is it because compute would not do anything new that persist has already started doing in the background before this?

Reply:
Yeah the task is already running as soon as we call `persist`.

Reply:
@jakirkham, the purpose of this here is to colocate the partitions when two dask arrays are passed in because they each need to be together on the same worker for training.

Reply:
Nice. I see `futures_of` can take a sequence of dask objects, but unfortunately, it flattens the futures into a single-dimensional list. Could have removed the next line if it returned a list of lists of futures of those objects.

Reply:
Yeah that's probably a consequence of [how `futures_of` extracts the `Future`s]( https://github.com/dask/distributed/blob/04de4b2adc1f0c94ab86e9aa46a4c67382a7eaea/distributed/client.py#L4447 ).

FWIW if you want to keep the structure, you could just grab the `.values()` from the graph as is done in that code (without the `.extend(...)` part).

Reply:
Could you please elaborate on the latter? I am still inexperienced with using dask and learning as I go.

Reply:
Hmm...now that I think about it. I'm not sure that would work anyways.